jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat Middleware LLC, and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.jpa.hibernate5.infinispan;
+
+import java.util.Properties;
+
+import org.hibernate.boot.spi.SessionFactoryOptions;
+import org.hibernate.cache.CacheException;
+
+/**
+ * Infinispan-backed region factory for use with standalone (i.e. non-JPA) Hibernate applications.
+ * @author Paul Ferraro
+ * @author Scott Marlow
+ * @deprecated Use {@link org.infinispan.hibernate.cache.v51.InfinispanRegionFactory} instead in conjunction with "hibernate.cache.infinispan.shared" set to false.
+ */
+@Deprecated
+public class InfinispanRegionFactory extends org.infinispan.hibernate.cache.v51.InfinispanRegionFactory {
+    private static final long serialVersionUID = 6526170943015350422L;
+
+    @Override
+    public void start(SessionFactoryOptions settings, Properties properties) throws CacheException {
+        properties.setProperty("hibernate.cache.infinispan.shared", "false");
```

Comment:
use `ManagedEmbeddedCacheManagerProvider.SHARED`?

Reply:
The module containing these classes intentionally does not reference the module containing ManagedEmbeddedCacheManagerProvider, which requires the jipijapa api, etc.  That's why I'm not using any constants here.
I could externalize this property to a separate file, in a separate module (similar to what you rightly did with InfinispanProperties), but that seemed overkill just for use by a compatibility module that will eventually be dropped.